test/models/VatObligationsSpec.scala

```diff
@@ -21,56 +21,208 @@ import play.api.libs.json.{JsValue, Json}
+        |				"periodKey": "17AA"
+        |			},
+        |			{
+        |				"status": "F",
+        |				"inboundCorrespondenceFromDate": "1981-02-03",
+        |				"inboundCorrespondenceToDate": "1981-04-05",
+        |				"inboundCorrespondenceDateReceived": "1981-02-02",
+        |				"inboundCorrespondenceDueDate": "1981-04-08",
+        |				"periodKey": "18AA"
+        |			}
+        |		]
+        |	}]
+        |}""".stripMargin)
+
+    val singleObligationOutputJson: JsValue = Json.parse(
+      """{
+        |	"obligations": [{
+        |		"obligationDetails": [{
+        |			"status": "F",
+        |			"start": "1980-02-03",
+        |			"end": "1980-04-05",
+        |			"received": "1980-02-02",
+        |			"due": "1980-04-08",
+        |			"periodKey": "17AA"
+        |		}, {
+        |			"status": "F",
+        |			"start": "1981-02-03",
+        |			"end": "1981-04-05",
+        |			"received": "1981-02-02",
+        |			"due": "1981-04-08",
+        |			"periodKey": "18AA"
+        |		}]
+        |	}]
+        |}""".stripMargin)
+
+    val singleObligationsModel: VatObligations =
+      VatObligations(
+        Seq(VatObligation(
+          Seq(
+            ObligationDetail("F", "1980-02-03", "1980-04-05", Some("1980-02-02"), "1980-04-08", "17AA"),
+            ObligationDetail("F", "1981-02-03", "1981-04-05", Some("1981-02-02"), "1981-04-08", "18AA")
+          )
+        )
         )
       )
-      )
-    )
-
-  val vatObligationsJson: JsValue = Json.parse(
-    """{
-      |	"obligations": [{
-      |		"identification": {
-      |			"referenceNumber": "555555555",
-      |			"referenceType": "VRN"
-      |		},
-      |		"obligationDetails": [{
-      |				"status": "F",
-      |				"inboundCorrespondenceFromDate": "1980-02-03",
-      |				"inboundCorrespondenceToDate": "1980-04-05",
-      |				"inboundCorrespondenceDateReceived": "1980-02-02",
-      |				"inboundCorrespondenceDueDate": "1980-04-08",
-      |				"periodKey": "17AA"
-      |			},
-      |			{
-      |				"status": "F",
-      |				"inboundCorrespondenceFromDate": "1981-02-03",
-      |				"inboundCorrespondenceToDate": "1981-04-05",
-      |				"inboundCorrespondenceDateReceived": "1981-02-02",
-      |				"inboundCorrespondenceDueDate": "1981-04-08",
-      |				"periodKey": "18AA"
-      |			}
-      |		]
-      |	}]
-      |}""".stripMargin)
-
-
-  "VatObligations" should {
 
     "serialize to Json successfully" in {
-      Json.toJson(vatObligationsModel) shouldBe vatObligationsJson
+      Json.toJson(singleObligationsModel) shouldBe singleObligationOutputJson
     }
 
     "deserialize to a Transaction model successfully" in {
-      vatObligationsJson.as[VatObligations] shouldBe vatObligationsModel
+      singleObligationInputJson.as[VatObligations] shouldBe singleObligationsModel
     }
 
   }
 
+
+  "VatObligations with multiple obligations" should {
+
+    val multipleObligationInputJson: JsValue = Json.parse(
```

Comment:
Same comment here as in the HttpParserSpec and ControllerSpec regarding valid scenario.

Reply:
Same reply..